h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -345,13 +346,27 @@ public void computeCrossValidation() {
       // scores; compute gains/lifts
       cv_mainModelScores(N, mbs, cvModelBuilders);
 
-      // Step 8: Clean up potentially created temp frames
-      for (ModelBuilder mb : cvModelBuilders)
-        mb.cleanUp();
-
       _job.setReadyForView(true);
       DKV.put(_job);
+    } catch (Exception e) {
+      if (cvModelBuilders != null) {
+        Futures fs = new Futures();
+        // removing keys added during cv_makeFramesAndBuilders and cv_makeFramesAndBuilders
+        // need a better solution: part of this is done in cv_makeFramesAndBuilders but partially and only for its method scope
+        for (ModelBuilder mb : cvModelBuilders) {
+          DKV.remove(mb._parms._train, fs);
+          DKV.remove(mb._parms._valid, fs);
+          DKV.remove(Key.make(mb.getPredictionKey()), fs);
+          mb._result.remove(fs);
```

Comment:
Should the result really be removed when the user presses the "Stop" button? The current behavior is to keep the Model that was not finished.

Reply:
by the way, the outlined code only removes the CV models of incomplete models.
The incomplete model itself was actually automatically removed in `ModeBuilder#trainModel` since the original timeout fix

Reply:
Sure, let's discuss over webex. 

Behavior in AutoML and for the user could be different and we can discuss that.

What happens when the model builder runs out of time in a regular model building scenario?

It seems to me that eg. in GBM & DRF it will produce an exception:

        if (stop_requested()) throw new Job.JobCancelledException();

which would lead to model deletion in this PR.

To sum it up, we need to discuss what should happen to the model when
- the user stops the training manually (eg. in Flow UI),
- model builder runs out of time.

Reply:
> by the way, the outlined code only removes the CV models of incomplete models.
The incomplete model itself was actually automatically removed in ModeBuilder#trainModel since the original timeout fix

Okay, then we need to revisit that as well. The point of this discussion is to make sure we didn't break things because of the changes related to `max_runtime_secs`.